browser/ui/views/tabs/brave_tab_container.cc

```diff
@@ -356,5 +384,112 @@ void BraveTabContainer::PaintChildren(const views::PaintInfo& paint_info) {
   }
 }
 
+BrowserRootView::DropIndex BraveTabContainer::GetDropIndex(
+    const ui::DropTargetEvent& event) {
+  if (!base::FeatureList::IsEnabled(tabs::features::kBraveVerticalTabs) ||
+      !tabs::utils::ShouldShowVerticalTabs(
+          tab_slot_controller_->GetBrowser())) {
+    return TabContainerImpl::GetDropIndex(event);
+  }
+
+  bool is_vtab_expanded =
+      GetVerticalTabState(tab_slot_controller_->GetBrowser()) ==
+      VerticalTabStripRegionView::State::kExpanded;
+
+  // Force animations to stop, otherwise it makes the index calculation tricky.
+  CompleteAnimationAndLayout();
+
+  const int x = GetMirroredXInView(event.x());
+  const int y = event.y();
+
+  std::vector<TabSlotView*> views = layout_helper_->GetTabSlotViews();
+
+  // Loop until we find a tab or group header that intersects |event|'s
+  // location.
+  for (TabSlotView* view : views) {
+    const int max_y = view->y() + view->height();
+    const int max_x = view->x() + view->width();
+    if (y >= max_y) {
+      continue;
+    }
+
+    if (view->GetTabSlotViewType() == TabSlotView::ViewType::kTab) {
+      Tab* const tab = static_cast<Tab*>(view);
+      // When the vertical tab strip state is expanded, pinned tabs are
+      // organized in a grid view. Therefore, it's important to also consider
+      // the x-axis location.
+      if (is_vtab_expanded && tab->data().pinned && x >= max_x) {
+        continue;
+      }
+
+      // Closing tabs should be skipped.
+      if (tab->closing()) {
+        continue;
+      }
+
+      const int model_index = GetModelIndexOf(tab).value();
+      const bool first_in_group =
+          tab->group().has_value() &&
+          model_index == controller_->GetFirstTabInGroup(tab->group().value());
+
+      // When hovering over the left or right quarter of a tab, the drop
+      // indicator will point between tabs.
+      const int hot_height = tab->height() / 4;
```

Comment:
Interestingly, when it's pinned tab, dropping a url between tabs doesn't create pinned tabs. So I think we don't need |hot_height| in case it's pinned tab.